docs/contributing/javaagent-jar-components.md

```diff
@@ -1,102 +0,0 @@
-from different libraries, it is paramount to hide all these classes from the
-host application. This is achieved in the following way:
-
-- When `javaagent` module builds the final agent, it moves all classes from
-`instrumentation` submodules, `javaagent-tooling` and `javaagent-extension-api` modules
-into a separate folder inside final jar file, called`inst`.
-In addition, the extension of all class files is changed from `class` to `classdata`.
-This ensures that general classloaders cannot find nor load these classes.
-- When `io.opentelemetry.javaagent.bootstrap.AgentInitializer` is invoked, it creates an
-instance of `io.opentelemetry.javaagent.bootstrap.AgentClassLoader`, loads an
-`io.opentelemetry.javaagent.tooling.AgentInstaller` from that `AgentClassLoader`
-and then passes control on to the `AgentInstaller` (now in the
-`AgentClassLoader`). The `AgentInstaller` then installs all of the
-instrumentations with the help of ByteBuddy. Instead of using agent classloader all agent classes
-could be shaded and used from the bootstrap classloader. However, this opens de-serialization
-security vulnerability and in addition to that the shaded classes are harder to debug.
-
-The complicated process above ensures that the majority of
-auto-instrumentation agent's classes are totally isolated from application
-classes, and an instrumented class from arbitrary classloader in JVM can
-still access helper classes from bootstrap classloader.
-
-### Agent jar structure
-
-If you now look inside
-`javaagent/build/libs/opentelemetry-javaagent-<version>.jar`, you will see the
-following "clusters" of classes:
-
-Available in the system class loader:
-
-- `io/opentelemetry/javaagent/bootstrap/AgentBootstrap` - the one class from `javaagent`
-module
-
-Available in the bootstrap class loader:
-
-- `io/opentelemetry/javaagent/bootstrap/` - contains the `javaagent-bootstrap` module
-- `io/opentelemetry/javaagent/instrumentation/api/` - contains the `javaagent-instrumentation-api` module
-- `io/opentelemetry/javaagent/shaded/instrumentation/api/` - contains the `instrumentation-api` module,
- shaded during creation of `javaagent` jar file by Shadow Gradle plugin
-- `io/opentelemetry/javaagent/shaded/io/` - contains the OpenTelemetry API and its dependency gRPC
-Context, both shaded during creation of `javaagent` jar file by Shadow Gradle plugin
-- `io/opentelemetry/javaagent/slf4j/` - contains SLF4J and its simple logger implementation, shaded
-during creation of `javaagent` jar file by Shadow Gradle plugin
-
-Available in the agent class loader:
-- `inst/` - contains `javaagent-tooling` and `javaagent-extension-api` modules and
-  `instrumentation` submodules, loaded and isolated inside `AgentClassLoader`.
-  Includes the OpenTelemetry SDK.
-
-![Agent initialization sequence](initialization-sequence.svg)
-[Image source](https://docs.google.com/drawings/d/1GHAcJ8AOaf_v2Ip82cQD9dN0mtvSk2C1B11KfwV2U8o)
```

Comment:
I removed the initialization sequence doc from the new version of this page: it was completely out of date, and I don't think it was that useful -- the overall structure/architecture of the javaagent is very useful for a contributor (I think? it's not like I have anything to back this up 😅 ), but the initialization sequence is something that is changed very rarely. If you really need to know you can just read the code, it's not that hard to follow (unlike the structure, where you need to learn what every top-level module does, and which CL it's loaded by)